src/main/java/de/tum/in/www1/artemis/config/StaticCodeAnalysisConfigurer.java

```diff
@@ -0,0 +1,90 @@
+package de.tum.in.www1.artemis.config;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.io.support.ResourcePatternUtils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.tum.in.www1.artemis.domain.StaticCodeAnalysisConfiguration;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+
+/**
+ * Reads static code analysis configurations from json files at application start-up and provides a Bean with
+ * programming language specific default categories.
+ */
+@Configuration
+public class StaticCodeAnalysisConfigurer {
+
+    private final Logger log = LoggerFactory.getLogger(StaticCodeAnalysisConfigurer.class);
+
+    private final ResourceLoader resourceLoader;
+
+    private final Map<ProgrammingLanguage, StaticCodeAnalysisConfiguration> languageToConfiguration = new HashMap<>();
+
+    public StaticCodeAnalysisConfigurer(ResourceLoader resourceLoader) {
+        this.resourceLoader = resourceLoader;
+    }
+
+    @PostConstruct
+    private void init() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String configurationsPath = "classpath:templates/staticCodeAnalysis/*.*";
+        Resource[] jsonConfigurations;
+
+        try {
+            jsonConfigurations = ResourcePatternUtils.getResourcePatternResolver(resourceLoader).getResources(configurationsPath);
+        }
+        catch (IOException e) {
+            log.debug("Could not load static code analysis configuration resources: " + e.getMessage());
+            return;
+        }
+
+        for (var jsonConfiguration : jsonConfigurations) {
```

Comment:
move everything into the try block and catch the generic Exception so that this code can never break the application start